src/FlaUI.Core/AutomationElements/TextBox.cs

```diff
@@ -48,10 +73,70 @@ public string Text
                 }
                 else
                 {
-                    Enter(value);
+                    if (SetTextWin32(value) == false)
+                    {
+                        Enter(value);
+                    }
                 }
             }
         }
+        
+        private bool SetTextWin32(string text)
+        {
+		// try with native Win32 function SetWindowText
+		if (Properties.NativeWindowHandle.IsSupported)
+		{
+			var windowHandle = Properties.NativeWindowHandle.ValueOrDefault;
+			if (windowHandle != IntPtr.Zero)
+			{
+				IntPtr textPtr = IntPtr.Zero;
+				try
+				{
+					textPtr = Marshal.StringToBSTR(text);
+				}
+				catch { }
+
+				try
+				{
+					if (textPtr != IntPtr.Zero)
+					{
+						if (SendMessage(windowHandle, WindowMessages.WM_SETTEXT, IntPtr.Zero, textPtr) ==
+							Win32Constants.TRUE)
+						{
+							return true; // text successfully set
+						}
+					}
+				}
+				catch { }
+				finally
```

Comment:
This finally should be part of the try above, where the object is created.

Reply:
It's ok where it is because I want to release the pointer no matter what. If I release it after the object is created I will use a released pointer. See the comments of my latest commit to TextBox.cs.